### PR TITLE
Added Blackmailed.com, VogoV.com, UltraFilms.xxx | Bugfixes

### PR DESCRIPTION
**Pull request by kamuk90**
- Added blackmailed.com site to list (on linux tested)
for searching code use the sitemap.
for success finding the scene you need to enter at least one word contained in the name of the scene
> ex. Blackmailed-Secretary's Blackmailed Office Fuck its same like Blackmailed-Secretary's

- Added VogoV.com site to list  (on linux tested)
Actors and title searching available

- Added Ultrafilms.xxx site to list  (on linux tested)
Actors and title searching available

- Mylf search fixed
- Letsdoit fix

### CODE REVIEW

File: Contents/Code/siteUltrafilmsxxx.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    try:
+        searchResults = HTML.ElementFromURL(url)
+    except:
+        request = urllib.Request(url, headers=headers)
+        response = urllib.urlopen(request, context=ssl.SSLContext(ssl.PROTOCOL_TLSv1))
+        htmlstring = response.read()
+        searchResults = fromstring(htmlstring)
+
+    for searchResult in searchResults.xpath('//ul[@class="listing-videos listing-tube"]'):
```
Comment (ferengi82) [inline Contents/Code/siteUltrafilmsxxx.py]: thanks for the ssl workaround!
but the xpath dont work for me, try this:

`    for searchResult in searchResults.xpath('//*[div[@class="format-infos"]/div[@class="time-infos"]]'):
        titleNoFormatting = searchResult.xpath('.//a')[0].get('title').strip()
        Log("titleNoFormatting: " + titleNoFormatting)
        actors = searchResult.xpath('.//img')[0].get('alt')
        Log(actors)
        sceneUrl = searchResult.xpath('.//a')[0].get('href')
        curID = sceneUrl.replace('/','_').replace('?','!')
        Log("curID: " + curID)
        score = 100 - Util.LevenshteinDistance(searchTitle.lower(), titleNoFormatting.lower())
        results.Append(MetadataSearchResult(id = curID + "|" + str(siteNum), name = titleNoFormatting + " with "+ actors + " [" + PAsearchSites.getSearchSiteName(siteNum) + "] ", score = score, lang = lang))
    return results`

File: Contents/Code/siteUltrafilmsxxx.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        pass
+
+    # Collections / Tagline
+    metadata.collections.clear()
+    metadata.tagline = siteName
+    metadata.collections.add(siteName)
+
+    # Genres
+    try:
+        genres = detailsPageElements.xpath('//div[@itemprop="keywords"]//ul//li')
```
Comment (ferengi82) [inline Contents/Code/siteUltrafilmsxxx.py]: and this

genres = detailsPageElements.xpath('//div[@itemprop="keywords"]//ul//li//a')

**Comment by ferengi82** [comment]
i have tested further: in some cases the searchresult appears not on the first Site
https://www.ultrafilms.xxx/page/2/?s=love+me+harder => Site 2 of 27.
so we have to walk trough all sites until wie find the searched title (or is there a better solution?)

**Comment by ferengi82** [comment]
at the moment im trying this, it works great, but only for scenes ...

`def search(results,encodedTitle,title,searchTitle,siteNum,lang,searchByDateActor,searchDate,searchSiteID):
    if searchSiteID != 9999:
        siteNum = searchSiteID
    url = 'https://www.ultrafilms.xxx/category/movies/'
    headers = {'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_9_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/35.0.1916.47 Safari/537.36'}

    try:
        searchResults = HTML.ElementFromURL(url)
    except:
        request = urllib.Request(url, headers=headers)
        response = urllib.urlopen(request, context=ssl.SSLContext(ssl.PROTOCOL_TLSv1))
        htmlstring = response.read()
        searchResults = fromstring(htmlstring)

    
    # getting movie page count

    pagecount = searchResults.xpath('//ul[@class="page-numbers"]/li[position() = (last() - 1)]')[0].text_content().strip()
    Log("Page Count: " + pagecount)

    for page in range(1,int(pagecount)+1):
        pageurl = 'https://www.ultrafilms.xxx/category/movies/page/' + str(page)
        Log("Actual Page: " + pageurl)
        
        try:
            searchResults = HTML.ElementFromURL(pageurl)
        except:
            request = urllib.Request(pageurl, headers=headers)
            response = urllib.urlopen(request, context=ssl.SSLContext(ssl.PROTOCOL_TLSv1))
            htmlstring = response.read()
            searchResults = fromstring(htmlstring)

        for searchResult in searchResults.xpath('//*[div[@class="format-infos"]/div[@class="time-infos"]]'):
            titleNoFormatting = searchResult.xpath('.//a')[0].get('title').strip()
            Log("titleNoFormatting: " + titleNoFormatting)
            actors = searchResult.xpath('.//img')[0].get('alt')
            Log(actors)
            sceneUrl = searchResult.xpath('.//a')[0].get('href')
            curID = sceneUrl.replace('/','_').replace('?','!')
            Log("curID: " + curID)
            score = 100 - Util.LevenshteinDistance(searchTitle.lower(), titleNoFormatting.lower())
            results.Append(MetadataSearchResult(id = curID + "|" + str(siteNum), name = titleNoFormatting + " with "+ actors + " [" + PAsearchSites.getSearchSiteName(siteNum) + "] ", score = score, lang = lang))
    return results`

**Comment by kamuk90** [comment]
Thanks dude, unfortunately I'm on vacation, save this code, I will test and add it to the next update

**Comment by ferengi82** [comment]
> Thanks dude, unfortunately I'm on vacation, save this code, I will test and add it to the next update

enjoy your vacation!

**Comment by claygoldfinch** [comment]
https://www.ultrafilms.xxx/?s=love+me+harder
vs
https://www.ultrafilms.xxx/?s=%22Love+Me+Harder%22
[https://www.ultrafilms.xxx/?s="Love+Me+Harder"](https://www.ultrafilms.xxx/?s="Love+Me+Harder")

**Comment by claygoldfinch** [comment]
I'm going to close this for this time being as it appears to be a WIP. Feel free to resubmit as a draft.

**Comment by ferengi82** [comment]
i changed the code based on claygoldfich's suggestions, and it seems to work fine.
[new code](https://gist.github.com/ferengi82/cbb5432dadb6b0b5f792c9b099c36b5e)

**Pull request closed without merging**